include/boost/config/compiler/xlcpp_zos.hpp

```diff
@@ -0,0 +1,201 @@
+//  Copyright (c) 2017 Dynatrace
+//
+//  Distributed under the Boost Software License, Version 1.0.
+//  See accompanying file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt
+
+//  See http://www.boost.org for most recent version.
+
+//  Compiler setup for IBM z/OS XL C/C++ compiler.
+
+// Oldest compiler version currently supported is 2.1 (V2R1)
+#if !defined(__IBMCPP__) || !defined(__COMPILER_VER__) || __COMPILER_VER__ < 0x42010000
+#  error "Compiler not supported or configured - please reconfigure"
+#endif
+
+#if __COMPILER_VER__ > 0x42010000
+#  if defined(BOOST_ASSERT_CONFIG)
+#     error "Unknown compiler version - please run the configure tests and report the results"
+#  endif
+#endif
+
+#define BOOST_COMPILER "IBM z/OS XL C/C++ version " BOOST_STRINGIZE(__COMPILER_VER__)
+#define BOOST_XLCPP_ZOS __COMPILER_VER__
+
+// -------------------------------------
+
+#include <features.h> // For __UU, __C99, __TR1, ...
+
+#if !defined(BOOST_XLCPP_ZOS_DO_NOT_GUESS)
+#  if (defined(__C99_MIXED_STRING_CONCAT) \
+    && defined(__C99_MAX_LINE_NUMBER) \
+    && defined(__C99_EMPTY_MACRO_ARGUMENTS) \
+    && defined(__C99_LLONG) \
+    && defined(__IBMCPP_C99_PREPROCESSOR) \
+    && defined(__IBMCPP_C99_LONG_LONG) \
+    && defined(__IBM_CHAR32_T__) \
+    && defined(__IBM_CHAR16_T__) \
+    && defined(__IBMCPP_VARIADIC_TEMPLATES) \
+    && defined(__IBMCPP_STATIC_ASSERT) \
+    && defined(__IBMCPP_SCOPED_ENUM) \
+    && defined(__IBMCPP_RVALUE_REFERENCES) \
+    && defined(__IBMCPP_RIGHT_ANGLE_BRACKET) \
+    && defined(__IBMCPP_REFERENCE_COLLAPSING) \
+    && defined(__IBMCPP_INLINE_NAMESPACE) \
+    && defined(__IBMCPP_EXTENDED_FRIEND) \
+    && defined(__IBMCPP_EXPLICIT_CONVERSION_OPERATORS) \
+    && defined(__IBMCPP_DELEGATING_CTORS) \
+    && defined(__IBMCPP_DECLTYPE) \
+    && defined(__IBMCPP_CONSTEXPR) \
+    && defined(__IBMCPP_AUTO_TYPEDEDUCTION) \
+   )
+//   According to documentation, the "defaulted and deleted functions" feature should be testable via the predefined macro __IBMCPP_DEFAULTED_AND_DELETED_FUNCTIONS.
+//   Unfortunately this doesn't work, the macro is actually never defined (compiler version V2R1).
+//   Guess: defaulted and deleted functions feature is active if ALL other C++11 features supported by the compiler are active.
+#    if !defined(__IBMCPP_DEFAULTED_AND_DELETED_FUNCTIONS)
+#      define __IBMCPP_DEFAULTED_AND_DELETED_FUNCTIONS 1
```

Comment:
If Boost.Config is to define a private macro, it would probably be better to use a private prefix for Boost.Config. `BOOST_CONFIG_DETAIL_` or something. I know it's long, that's why I suggested to define the public macros right away.